plutarch-test/src/Plutarch/Test/Golden.hs

```diff
@@ -1,8 +1,11 @@
+{-# LANGUAGE ImpredicativeTypes #-}
```

Comment:
Hard dependency on GHC 8?

Reply:
I meant to write GHC 9. If you don't care about GHC 8, then sure, but I think we still do a bit. Anyway, just wrap the inner `ClosedTerm` in a newtype.

Reply:
`Identity (ClosedTerm a)` might be enough.

Reply:
otherwise make `newtype ClosedTerm' a = ClosedTerm' { unClosedTerm' :: ClosedTerm a }`

Reply:
@L-as I'm not sure I understand. CI for GHC 8.10 passes, therefore this change is still supported in GHC 8, I'd think? 

Since this code already compiles in GHC 8.10, what would be the advantage of wrapping in a newtype?